core/internal/filetransfer/file_transfer_default.go

```diff
@@ -78,7 +78,7 @@ func (ft *DefaultFileTransfer) Upload(task *Task) error {
 			if task.ProgressCallback != nil {
 				task.ProgressCallback(processed, total)
 			}
-
+			// TODO: this does not take errors into account
```

Comment:
@timoffex this can produce weird intermediate results as a file keeps getting re-read, like
uploadede 2.00 MB / 0.1 MB or something.

Reply:
oh, wow. That's a problem in the ProgressReader though, not in this function. It needs to update `processed` when `Seek` is used.